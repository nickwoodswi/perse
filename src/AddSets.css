.set-number-input{
    width: 40px;
    height: 70px;
}