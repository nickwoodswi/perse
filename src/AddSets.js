import React, { Component } from 'react'
import './AddSets.css'

class AddSets extends Component {
    static defaultProps = {
        targetSet: {}
      }
    render() {
        return(
            <>
            <label htmlFor="set-number-input">Number of Sets</label>
            <input id="set-number-input" className="set-number-input"
                type="number" 
                value={this.props.targetSet.set_num}
                onChange={e => this.props.addSets(this.props.setId, this.props.targetSet, e.target.value)}/>
            </>
        )
    }
}

export default AddSets