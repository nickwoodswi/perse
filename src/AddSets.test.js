import React from 'react';
import { render } from '@testing-library/react';
import AddSets from './AddSets';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders addsets component', () => {
    render(<MemoryRouter><AddSets /></MemoryRouter>);
});

it('AddSets component renders the UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><AddSets /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});