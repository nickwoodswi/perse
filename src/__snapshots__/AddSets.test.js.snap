// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSets component renders the UI as expected 1`] = `
<input
  className="set-number-input"
  onChange={[Function]}
  type="number"
/>
`;
